frontend/src/app/projects/models/create-coworking-method/create-coworking-method.component.ts

```diff
@@ -13,58 +20,139 @@ import {
+    branches: [],
+    tags: [],
+  };
+  public filteredRevisions: Instance = { branches: [], tags: [] };
+
+  public resultUrl: string = '';
 
   public form = new FormGroup({
+    urlGroup: new FormGroup({
+      baseUrl: new FormControl<GitSetting | undefined>(undefined),
+      inputUrl: new FormControl('', absoluteUrlSafetyValidator()),
+    }),
     credentials: new FormGroup({
-      path: new FormControl('', Validators.required),
       username: new FormControl(''),
       password: new FormControl(''),
     }),
-    revision: new FormControl('', Validators.required),
+    revision: new FormControl('', [
+      Validators.required,
+      this.validRevisionValidator(),
+    ]),
     entrypoint: new FormControl('/'),
   });
 
-  public filteredRevisions: Instance = { branches: [], tags: [] };
-
   constructor(
     public projectService: ProjectService,
     public modelService: ModelService,
+    public gitSettingsService: GitSettingsService,
     private gitService: GitService,
     private sourceService: SourceService
   ) {}
 
   ngOnInit(): void {
-    this.form.controls.revision.valueChanges.subscribe((value) => {
-      if (!this.gitService.instance) {
-        this.filteredRevisions = { branches: [], tags: [] };
-      } else {
-        this.filteredRevisions = {
-          branches: this.gitService.instance.branches.filter((branch) =>
-            branch.startsWith(value as string)
-          ),
-          tags: this.gitService.instance.tags.filter((tag) =>
-            tag.startsWith(value as string)
-          ),
-        };
+    this.gitService.instance.subscribe((revisions) => {
+      this.availableRevisions = revisions;
+      this.form.controls.revision.updateValueAndValidity({
+        onlySelf: false,
+        emitEvent: true,
+      });
+    });
+
+    this.gitSettingsService.gitSettings.subscribe((gitSettings) => {
+      this.availableGitSettings = gitSettings;
+
+      if (gitSettings.length) {
+        this.form.controls.urlGroup.controls.baseUrl.setValidators([
+          Validators.required,
+        ]);
+        this.form.controls.urlGroup.controls.inputUrl.setValidators([
+          absoluteOrRelativeSafetyValidators(),
+        ]);
       }
     });
+
+    this.form.controls.revision.valueChanges.subscribe((value) =>
+      this.filteredRevisionsByPrefix(value as string)
+    );
+
+    this.form.controls.urlGroup.setValidators([this.resultUrlValidator()]);
+    this.gitSettingsService.loadGitSettings();
   }
 
   onRevisionFocus(): void {
-    this.gitService
-      .fetch('', this.form.controls.credentials.value as Credentials)
-      .subscribe((instance) => {
-        this.filteredRevisions = instance;
-      });
+    let urlGroup = this.form.controls.urlGroup;
+    if (
+      urlGroup.invalid ||
+      urlGroup.controls.baseUrl.invalid ||
+      urlGroup.controls.inputUrl.invalid
+    ) {
+      return;
+    }
+
+    let projectName = this.projectService?.project?.name || '';
```

Comment:
```suggestion
    let projectName = this.projectService!.project!.name;
```

If you expect that a project is there, use the `!` instead of the `?`.